ocaml-lsp-server/src/hover_req.ml

```diff
@@ -12,33 +12,63 @@ let environment_mode =
   | Some true -> Extended_variable
   | Some false | None -> Default
 
-let format_contents ~syntax ~markdown ~typ ~doc =
+let print_dividers sections =
+  match sections with
+  | [] -> ""
+  | _ ->
+    String.concat ~sep:"\n\n --- \n\n" sections
+
+let format_as_code_block strings =
+  match strings with
+  | [] -> ""
+  | _ -> sprintf "```\n %s \n```" (String.concat ~sep:" " strings)
```

Comment:
You're not formatting the code blocks the same as before (not sure if the extra spaces have an impact however).
But more importantly this function should have two arguments: the name and the type, which are both strings. No need for a list here given the usage of the function.